aws-s3-public-bucket/main.tf

```diff
@@ -0,0 +1,82 @@
+locals {
+  tags = {
+    project         = var.project
+    env             = var.env
+    service         = var.service
+    owner           = var.owner
+    managedBy       = "terraform"
+    isPublic        = true
+    public_reason   = var.public_reason
+    bucket_contents = var.bucket_contents
+  }
+
+  // make sure we have public in the name
+  bucket_name = replace(var.bucket_name, "public", "") == var.bucket_name ? "${var.bucket_name}-public" : var.bucket_name
+}
+
+resource "aws_s3_bucket" "bucket" {
+  bucket = local.bucket_name
+
+  policy = data.aws_iam_policy_document.bucket_policy.json
+
+  versioning {
+    enabled = true
+  }
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+  tags = local.tags
+}
+
+data "aws_iam_policy_document" "bucket_policy" {
+  # Deny access to bucket if it's not accessed through HTTPS
+  source_json = var.bucket_policy
+
+  statement {
+    sid       = "EnforceHTTPS"
+    actions   = ["s3:GetObject"]
+    resources = ["arn:aws:s3:::${local.bucket_name}/*"]
+
+    principals {
+      type        = "*"
+      identifiers = ["*"]
+    }
+
+    effect = "Deny"
+
+    condition {
+      test     = "Bool"
+      variable = "aws:SecureTransport"
+      values   = ["false"]
+    }
+  }
```

Comment:
I don't know we can enforce this statement on all public read buckets since (last I knew) S3 websites don't allow for tls

Reply:
Deleted this statement